src/app-layout/__tests__/runtime-drawers.test.tsx

```diff
@@ -290,6 +290,37 @@ describeEachAppLayout(({ theme, size }) => {
     expect(onToolsChange).toHaveBeenCalledWith({ open: false });
   });
 
+  test('respect controlled toolsOpen with runtime drawers after clicking on tools drawer', async () => {
```

Comment:
This fix also resolves an additional issue with partially controllable tools (`toolsOpen` is provided while `onToolsChange` not), such as in the following case:

```js
<AppLayout tools="Tools content" toolsOpen={false} />
```

Here, clicking the toolbar trigger should not open the tools panel. Let's add a unit test to cover this scenario as well